apm-agent-plugins/apm-hibernate-search-plugin/apm-hibernate-search-plugin-5_x/src/main/java/co/elastic/apm/agent/hibernate/search/v5_x/HibernateSearch5Instrumentation.java

```diff
@@ -0,0 +1,82 @@
+package co.elastic.apm.agent.hibernate.search.v5_x;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import javax.annotation.Nullable;
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation;
+import co.elastic.apm.agent.bci.VisibleForAdvice;
+import co.elastic.apm.agent.hibernate.search.HibernateSearchConstants;
+import co.elastic.apm.agent.impl.transaction.Span;
+import co.elastic.apm.agent.impl.transaction.TraceContextHolder;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.hibernate.search.query.hibernate.impl.FullTextQueryImpl;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+public class HibernateSearch5Instrumentation extends ElasticApmInstrumentation {
+
+    @Override
+    public Class<?> getAdviceClass() {
+        return HibernateSearch5ExecuteAdvice.class;
+    }
+
+    @Override
+    public ElementMatcher<? super TypeDescription> getTypeMatcher() {
+        return named("org.hibernate.search.query.hibernate.impl.FullTextQueryImpl");
+    }
+
+    @Override
+    public ElementMatcher<? super MethodDescription> getMethodMatcher() {
+        return named("list");
+    }
+
+    @Override
+    public Collection<String> getInstrumentationGroupNames() {
+        return Collections.singleton(HibernateSearchConstants.HIBERNATE_SEARCH_ORM_TYPE);
+    }
+
+    @VisibleForAdvice
+    public static class HibernateSearch5ExecuteAdvice {
+
+        @Advice.OnMethodEnter(suppress = Throwable.class)
+        private static void onBeforeExecute(@Advice.This FullTextQueryImpl query,
+            @Advice.Local("span") Span span) {
+            if (tracer != null) {
+                TraceContextHolder<?> active = tracer.getActive();
+                if (active == null || active instanceof Span && HibernateSearchConstants.HIBERNATE_SEARCH_ORM_TYPE
+                    .equals(((Span) active).getSubtype())) {
+                    return;
+                }
+
+                final @Nullable String queryString = query.getQueryString();
+                span = active.createSpan().activate();
+
+                span.withType("db")
```

Comment:
I am not sure about this being a `db` type. If it is, maybe we should consider more basic hibernate stuff? Need to think a bit more about this 🤔

Reply:
I also wasn't really sure about this 😉 The `db` type was the closest type already used by other plugins I found.

Reply:
We _DO_ use `db` as type for the Elasticsearch client, so we can stick to it right now, but may reconsider later.